src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -446,6 +451,21 @@ def run(self, batch_inputs):
         result_list = []
         run_start_time = datetime.utcnow()
 
+        # In fork mode, the spawned process is the bridge between the main and fork processes.
+        # If the spawned process is no longer running, exit the main proccess.
+        if self._use_fork:
```

Comment:
The idea looks good to me.
As the implmenetation, I prefer the proposal like this:
`self._process_manager.ensure_healthy()` then all related logic could be moved to the implementation of the process manager instead of `if else` in the process_pool logic.